### PR TITLE
If a user hasn't interacted with the camera, continue to resize to the orbit-eye

### LINKED ISSUE #2829: Interpolate the camera to accomodate the scene's bbox
We continuously recompute the bounding box encompassing the scene, but only move the camera once at the "start" (which is a racy definition by nature, which is part of the problem).

We should move the camera continuously (interpolating smoothly between positions) too as the bounding box changes, as long as the user hasn't modified it manually.

### PR DESCRIPTION
**Pull request by jleibs**
### What
Our initial orbit-eye state is sensitive to the timing of what content is on screen when the scene is initialized.

There appears to be maybe be a regression that stems from *faster* scene initialization, potentially interacting pathologically with the fact that our blueprint doesn't yet store the camera position. In any case, there are easy-to-reproduce situations where our timeline hasn't yet updated from the initial time when the scene is first populated, meaning many scenes just end up with the empty-scene initizalization for the orbit-eye, and then stick there even as content is added to the scene.

This only appears to affect examples that run live via `.connect()`. Using the viewer with a pre-saved rrd file (either locally or on the web) mitigates the issue -- presumably because this causes our timestamp to be updated sooner.

However, even if we initialize timeline differently relative to view creation, it's very easy to create scenarios where you log a small piece of content followed by a large piece of content which ends up significantly outside the FOV.

### The Solution

We now track whether any user interaction has occured, and if it hasn't continue to interpolate to the new target eye, bringing new content into the field of view of the scene.

### Examples of worse behavior
It's worth noting the behavior of arkitscenes does seem like a regression relative to the current behavior. I don't love the scene bouncing around that way.

https://demo.rerun.io/pr/2839/examples/arkit_scenes/

### Examples of improved behavior
However, the behavior of linetrip example and colmap were arguably totally broken before and this seems like a huge improvement:

```
python docs/code-examples/line_strip3d_batch.py
```
Before   |  After
:----:|:-----:
image | image

```
python examples/python/structure_from_motion/main.py --dataset colmap_fiat
```
Before   |  After
:----:|:-----:
image | image

Fixes #2829 

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/2839) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/2839)
- [Docs preview](https://rerun.io/preview/pr%3Ajleibs%2Fgrowing_eye/docs)
- [Examples preview](https://rerun.io/preview/pr%3Ajleibs%2Fgrowing_eye/examples)

### CODE REVIEW

**Comment by Wumpf** [comment]
> It's worth noting the behavior of arkitscenes does seem like a regression relative to the current behavior. I don't love the scene bouncing around that way.

I see the bouncing but I don't find it all that irritating as it is smooth enough.
It's interesting though that the camera is as far out as it is; might be worth tweaking / looking into why it is that big.
We probably should at least for debug purposes visualize both the bounding box and the accumulated bounding box

File: crates/re_space_view_spatial/src/ui_3d.rs [hunk 1]
```diff
@@ -92,6 +94,24 @@ impl View3DState {
...
+                    elapsed_time: 0.0,
+                    target_time,
+                    start,
+                    target_orbit: Some(target),
+                    target_eye: None,
+                });
+            } else {
+                self.orbit_eye = Some(target);
+            }
+        }
```
Comment (Wumpf) [inline crates/re_space_view_spatial/src/ui_3d.rs:113]: Don't understand the borrow concern. this just works for me
```suggestion
        if !self.did_interact_with_eye {
            let target = default_eye(scene_bbox_accum, &view_coordinates);
            self.interpolate_to_orbit_eye(target);
        }
```

**Review by Wumpf — APPROVED** [review]
played around with it and I really like the behavior but I'd be advocating for having the default_eye generally be further zoomed in while we're on it

**Comment by Wumpf** [comment]
Let's talk about this briefly in the daily

File: crates/re_space_view_spatial/src/ui_3d.rs (hunk 1 above)
Comment (teh-cmc) [inline crates/re_space_view_spatial/src/ui_3d.rs:113]: That doesn't seem necessary anymore.
```suggestion
        if !self.did_interact_with_eye {
            self.interpolate_to_orbit_eye(default_eye(scene_bbox_accum, &view_coordinates));
        }
````

Comment (Wumpf) [reply crates/re_space_view_spatial/src/ui_3d.rs:113]: I was first :D

**Comment by Wumpf** [comment]
It seems we still have a bug in the bounding box computation somewhere - it ends up being to big occasionally, causing the too-away camera I'm complaining about

**Comment by Wumpf** [comment]
the biggest risk is that if the user has a single point _really_ far way, the camera will be bonkers. But this is a *pre-existing* problem as you'd also get that problematic camera when you reset camera (via double click)

**Comment by Wumpf** [comment]
Experimented with closer cameras and it doesn't work for the general case, what we have is about as good as it can be unless we start reasoning about "visual weight" of a scene

**Pull request merged**